### DISCUSSION TITLE
Odroid N2+ Not Booting With Fresh Install

### DISCUSSION THREAD

**Discussion by scoobydrvr** [home-assistant/operating-system#3226] [Q&A]
### Problem Statement and Experience
Purchased an Odroid-N2+ to replace my Raspberry Pi as the HA host; choosing the [4Gb version](https://www.hardkernel.com/shop/odroid-n2-with-4gbyte-ram-2/) with a 32Gb eMMC card. Initially, I tried to follow [the official instructions](https://www.home-assistant.io/common-tasks/os/#flashing-an-odroid-n2) for writing the HA image using the USB OTG interface but, unfortunately, I think my lone USB micro cable is failing as the verify step consistently failed. Belena seemed to have no issues writing the image but the "storage device" presented by the Odroid would disconnect and reconnect during the verify stage.
Having no confidence the image was being written correctly, I used the [mmc reader from Hardkernel](https://www.hardkernel.com/shop/usb3-0-emmc-module-writer-2/) and Belena to write the HA OS image to the card. Ensuring that the boot mode switch was toggled to "MMC" I connected the board to network and power and waited for the system to boot. After a brief period of time, the device appeared in my network admin console with a valid IP address but, after waiting as long as a half hour, the address continues to give me a "The connection has timed out" error message. There are some indications that _things_ are happening - both network status lights are illuminated on the network port (green solid, orange intermittently and briefly flashing off); the red power LED is on; and it responds to a ping; however, the flashing "heartbeat" blue LED does not occur.

### Troubleshooting Steps Performed
To ensure basic functionality of the board, I flashed the Ubuntu image from Hardkernel, using the same process as before, and connected keyboard and monitor. As expected, the device boots to the operating system and shows the "heartbeat" flashing blue LED.
Thinking perhaps there is an issue with HA OS 12.0, I tried the same steps using OS 11.5 but the issue persisted with the same behavior.
Hoping to get some sign of life with the HA image, I switched the boot selector to "SPI" and connected keyboard and monitor, expecting to see the Petitboot Terminal. This resulted in the network port only illuminating the orange light and the monitor saying "No input", but the flashing "heartbeat" pattern of the blue LED resumed.
Considering the possibility that my existing HA instance might be clashing with the hostname HA brings out of the box, I renamed the current one to "homeassisant-old". After this change, I re-flashed the image for the Odroid and attempted to boot again with no change in behavior. I've even completely disconnected the Pi from the network and started from the beginning with no success.

At this point, I'm stumped, but it appears [others are having problems as well](https://community.home-assistant.io/t/haos-want-boot-on-odroid-n2-after-fresh-install/546080). Can someone please assist?

**Comment by scoobydrvr** [comment]
Well, out of desperation, I tried to start it with the HA OS 11.4 image and, to my surprise, everything fired up! I get the heartbeat pattern from the blue LED and had a welcome screen waiting for me in fairly short order. I guess I’ll be submitting an issue here. Hopefully this helps someone in the future!

### Followup
After creating a user account and setting up the basic info, HA informed me, as expected, that there is an update to be applied. Once the update for 11.4 kicked off, the hardware returned to its previous behavior so there is clearly a difference between 11.3 and 11.4 that's causing it to hang. time to go compare files, I suppose...

**Comment by DavidRSullivan** [comment]
My main install started doing this on ODROID N2+, been running for years, and now wont boot.
Tried a fresh, 12.1, no go.
Tried a fresh, 12.0, no go.
Tried a fresh 11.5, no go.
Tried a linux disk to make sure hardware fine - all good.
Tried a fresh 11.4, no go.
Down to 8.5 no go.

Some logs:
[logs (1).txt](https://github.com/home-assistant/operating-system/files/14895385/logs.1.txt)
[logs.txt](https://github.com/home-assistant/operating-system/files/14895386/logs.txt)

Crazy is that it is getting "Buffer I/O Errors on mmcblk04" -- on a brand new SD card which I have put linux on, booted and fsck'd the drive just fine.

**Reply by cvasantrao** [reply]
Have you solved it? I have an identical issue. SDCard  failed and had to reinstall. Tried 12.4 but the setup screen never appeared. The system gets an IP address and I am able to ping. Nothing beyond that. Can you throw some light on how you resolved it?

**Comment by scoobydrvr** [comment]
Interesting that you're also having problems with an SD card... I opened an issue for my experience and it seems to be pointing to the size of the eMMC card. Might be unrelated, might be worth weighing in...
https://github.com/home-assistant/operating-system/issues/3227

**Reply by DavidRSullivan** [reply]
WIll do, I'm waiting for an eMMC and New ODROID N+ -- so I can test both a brand new and my previous 100% working N+.

Did the Industrial USB to Serial Adapter UART Serial Adapter FT232RL  you linked to in the other thread work?

**Reply by scoobydrvr** [reply]
Yes, thankfully!
